src/popup/pages/Settings.tsx

```diff
@@ -1,5 +1,151 @@
-import React from 'react'
+import React, { useState } from 'react'
+
+import { Input } from '@/components/input'
+import { Label } from '@/components/label'
+import { RadioGroup } from '@/components/radio-group'
+import { Button } from '@/components/button'
+import { DollarSign, WarningSign } from '@/components/icons'
+
+import { usePopup } from '@/providers/popup.state'
 
 export const Settings = () => {
-  return <>Settings</>
+  const {
+    data: {
+      connected,
+      wmEnabled,
+      wallet,
+      amount,
+      amountType: { recurring },
+    },
+    setData,
+  } = usePopup()
+
+  const [_wallet, setWallet] = useState<string>(wallet)
+  const [_walletError, setWalletError] = useState<string>('')
+  const [_amount, setAmount] = useState<string>(`${amount}`)
+  const [_amountError, setAmountError] = useState<string>('')
+  const [_recurring, setRecurring] = useState<string>(`${recurring}`)
+  const [loading, setLoading] = useState<boolean>(false)
+
+  const handleChangeWallet = (event: any) => {
+    setWallet(event.target.value)
+  }
+
+  const handleChangeAmount = (event: any) => {
+    setAmount(event.target.value)
+  }
+
+  const handleChangeRecurring = (value: string) => {
+    setRecurring(value)
+  }
```

Comment:
Let's use a boolean type for the `recurring` value.

Reply:
input fields work with strings, casting all the time is useless, I would convert at the end when we submit the value